webpack/JobInvocationDetail/CheckboxesActions.js

```diff
@@ -0,0 +1,195 @@
+    const newWin = window.open(url);
+
+    if (!newWin || newWin.closed || typeof newWin.closed === 'undefined') {
+      setShowAlert(true);
+    }
+  };
+
+  const fetchFailedHosts = async () => {
+    const { data } = await axios.get(
+      foremanUrl(`/api/job_invocations/${jobID}/hosts`),
+      {
+        params: {
+          per_page: MAX_HOSTS_API_SIZE,
+          search: `job_invocation.result = failed`,
+        },
+      }
+    );
+    return data?.results || [];
+  };
+
+  const handleOpenHosts = async (type = 'all') => {
+    if (type === 'failed') {
+      if (failedCount <= 3) {
+        const fetchedHosts = await fetchFailedHosts();
+        fetchedHosts.forEach(host =>
+          openLink(templateInvocationPageUrl(host.id, jobID))
+        );
+        return;
+      }
+      setIsOpenFailed(true);
+      setIsModalOpen(true);
+      return;
+    }
+
+    // type === 'all'
+    if (idsArray.length <= 3) {
+      idsArray.forEach(id => openLink(templateInvocationPageUrl(id, jobID)));
+      return;
+    }
+
+    setIsOpenFailed(false);
+    setIsModalOpen(true);
+  };
+
+  const ActionsKebab = () => {
+    const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+    const dropdownItems = [
+      <DropdownItem
+        ouiaId="open-failed-dropdown-item"
+        key="open-failed"
+        onClick={() => handleOpenHosts('failed')}
+        isDisabled={failedCount.length === 0}
+      >
+        {__(`Open all failed runs (${failedCount})`)}
+      </DropdownItem>,
+    ];
+
+    return (
+      <Dropdown
+        isOpen={isDropdownOpen}
+        onOpenChange={setIsDropdownOpen}
+        onSelect={() => setIsDropdownOpen(false)}
+        ouiaId="actions-kebab"
+        shouldFocusToggleOnSelect
+        toggle={toggleRef => (
+          <MenuToggle
+            aria-label="actions dropdown toggle"
+            id="toggle-kebab"
+            isExpanded={isDropdownOpen}
+            onClick={() => setIsDropdownOpen(prev => !prev)}
+            ref={toggleRef}
+            variant="plain"
+          >
+            <EllipsisVIcon />
+          </MenuToggle>
+        )}
+      >
+        <DropdownList>{dropdownItems}</DropdownList>
+      </Dropdown>
+    );
+  };
+
+  const OpenAllButton = () => (
+    <Button
+      aria-label="open all template invocations in new tab"
+      className="open-all-button"
+      isDisabled={idsArray.length === 0}
+      isInline
+      onClick={() => handleOpenHosts('all')}
+      ouiaId="template-invocation-new-tab-button"
+      variant="link"
+    >
+      <Tooltip content="Open selected in new tab">
+        <OutlinedWindowRestoreIcon />
+      </Tooltip>
+    </Button>
+  );
+
+  const RerunSelectedButton = () => (
+    <Button
+      aria-label="rerun selected template invocations"
+      className="rerun-selected-button"
+      component="a"
+      href={foremanUrl(
+        `/job_invocations/${jobID}/rerun?${idsArray
+          .map(id => `host_ids[]=${id}`)
+          .join('&')}`
```

Comment:
Does it make sense to use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to construct the query string? I think generally speaking it's a good practice. Here it may not matter that much, but it should deal well with escaping. `[]` should be escaped.